schema_based_on_diagram.sql

```diff
@@ -0,0 +1,48 @@
+CREATE TABLE patients (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    name VARCHAR,
+    date_of_birth DATE
+);
+
+CREATE TABLE invoices (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INTEGER REFERENCES medical_histories(id)
+);
+
+CREATE TABLE medical_histories (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patient_id INTEGER REFERENCES patients(id),
+    status VARCHAR
+);
+
+CREATE TABLE invoice_items (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity INTEGER,
+    total_price DECIMAL,
+    invoice_id INTEGER REFERENCES invoices(id),
+    treatment_id INTEGER REFERENCES treatments(id)
+);
+
+CREATE TABLE treatments (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    type VARCHAR,
+    name VARCHAR
+);
+
+CREATE TABLE medical_histories_treatments (
+    medical_history_id INTEGER REFERENCES medical_histories(id);
+    treatment_id INTEGER REFERENCES treatments(id)
+);
+
+CREATE INDEX medical_history_index ON medical_histories(medical_history_id);
+CREATE INDEX treatment_index ON treatments(treatment_id);
+CREATE INDEX patients_index ON patients(patient_id);
+CREATE INDEX invoice_index ON invoices(invoice_id); 
+CREATE INDEX medical_history_index ON medical_histories_treatments(medical_history_id);
+CREATE INDEX treatment_index ON medical_histories_treatments(treatment_id);
+
```

Comment:
- [ ]  I would like to recommend you work on the many-to-many relationships and implement all missing foreign keys as required for this project. As you can see on the screenshot below :arrow_down: Some all FOREIGN KEYS are not set. Please fix it :smile: 
![Screenshot from 2022-10-26 18-15-57](https://user-images.githubusercontent.com/6615394/198105222-1651ee54-ef97-43c8-823b-de6a2a91ace6.png)